test/integration/scheduler/preemption_test.go

```diff
@@ -366,6 +372,78 @@ func TestDisablePreemption(t *testing.T) {
 	}
 }
 
+func TestPodPriorityResolution(t *testing.T) {
+	admission := priority.NewPlugin()
+	context := initTestScheduler(t, initTestMaster(t, "preemption", admission), true, nil)
+	defer cleanupTest(t, context)
+	cs := context.clientSet
+
+	// Build clientset and informers for controllers.
+	externalClientset := kubernetes.NewForConfigOrDie(&restclient.Config{
+		QPS:           -1,
+		Host:          context.httpServer.URL,
+		ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	externalInformers := informers.NewSharedInformerFactory(externalClientset, time.Second)
+	admission.SetExternalKubeClientSet(externalClientset)
+	admission.SetExternalKubeInformerFactory(externalInformers)
+	externalInformers.Start(context.stopCh)
+	externalInformers.WaitForCacheSync(context.stopCh)
+
+	tests := []struct {
+		PriorityClass    string
+		Pod              *v1.Pod
+		ExpectedPriority int32
+	}{
+		{
+			PriorityClass:    scheduling.SystemNodeCritical,
+			ExpectedPriority: scheduling.SystemCriticalPriority + 1000,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod1-%v", scheduling.SystemNodeCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemNodeCritical,
+			}),
+		},
+		{
+			PriorityClass:    scheduling.SystemClusterCritical,
+			ExpectedPriority: scheduling.SystemCriticalPriority,
+			Pod: initPausePod(cs, &pausePodConfig{
+				Name:              fmt.Sprintf("pod2-%v", scheduling.SystemClusterCritical),
+				Namespace:         metav1.NamespaceSystem,
+				PriorityClassName: scheduling.SystemClusterCritical,
+			}),
+		},
+	}
+
+	// Create a node with some resources and a label.
+	nodeRes := &v1.ResourceList{
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+		v1.ResourceCPU:    *resource.NewMilliQuantity(500, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(500, resource.DecimalSI),
+	}
+	_, err := createNode(context.clientSet, "node1", nodeRes)
+	if err != nil {
+		t.Fatalf("Error creating nodes: %v", err)
+	}
+
+	pods := make([]*v1.Pod, len(tests))
+	for i, test := range tests {
+		pod, err := runPausePod(cs, test.Pod)
+		if err != nil {
+			t.Fatalf("Test [PodPriority/%v]: Error running pause pod: %v", test.PriorityClass, err)
+		}
+		pods[i] = pod
+		if pods[i].Spec.Priority != nil {
```

Comment:
This check doesn't make much sense. `initPausePod` already sets the `Spec.Priority`. What's the output we expect from the scheduler? It looks like everything is already covered by `runPausePod`.

Reply:
If I understand correctly, we are actually setting `PriorityClassName` (string) and verifying that it resolves to the expected `Priority` (int). So, part of the test is checking that `pod.Spec.Priority` is set by the priority admission plugin

Reply:
I see. Sounds good